provider/azure.go

```diff
@@ -222,6 +233,13 @@ func (p *AzureProvider) ApplyChanges(changes *plan.Changes) error {
 }
 
 func (p *AzureProvider) zones() ([]dns.Zone, error) {
+	// If we have the zones cached AND we have refreshed the cache in the last
+	// hour, THEN just return the cached results.
```

Comment:
Nit: comment is slightly inaccurate since it returns cached results until the interval elapses, rather than `last hour`.